services/DeviceInformation/include/ble-service-device-information/DeviceInformationService.h

```diff
@@ -0,0 +1,92 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_DEVICE_INFORMATION_SERVICE_H
+#define BLE_DEVICE_INFORMATION_SERVICE_H
+
+#include "ble/BLE.h"
+
+#if BLE_FEATURE_GATT_SERVER
+
+/** BLE Device Information Service class doesn't require instantiation - the characteristics added are read only.
+ * Use the static method add_service to add the chosen Device Information Service characteristics to the server. */
+class DeviceInformationService {
+public:
+    /** System ID */
+    struct SystemId {
```

Comment:
The convention in ble as been to use `snake_case_t` for pods see: https://os.mbed.com/docs/mbed-os/v6.9/contributing/style.html.

Reply:
Thanks, I'll rename them.

In my defence in BLE we also use CamelCase_t and CamelCase for pods.

Reply:
I think all the new types we defined in the recent year use `snake_case_t` (I might be wrong). It would be nice to have some alignment with general Mbed this time around. 

It think we should start a discussion on this item, Ble not being compliant with other Mbed OS guidelines...